resin/models/supervisor.py

```diff
@@ -0,0 +1,290 @@
+        RESIN_SUPERVISOR_API_KEY (str): supervisor API key on device.
+        _on_device (bool): API endpoint flag.
+            If True then all commands will be sent to the API on device.
+            If False then all command will be sent to the Resin API proxy endpoint (api.resin.io/supervisor/<url>).
+            If RESIN_SUPERVISOR_ADDRESS and RESIN_SUPERVISOR_API_KEY are available, _on_device will be set to True by default. Otherwise, it's False.
+
+    """
+
+    SUPERVISOR_API_VERSION = 'v1'
+    RESIN_SUPERVISOR_ADDRESS = os.environ.get('RESIN_SUPERVISOR_ADDRESS')
+    RESIN_SUPERVISOR_API_KEY = os.environ.get('RESIN_SUPERVISOR_API_KEY')
+
+    # _on_device = True, if RESIN_SUPERVISOR_ADDRESS and RESIN_SUPERVISOR_API_KEY env vars are avaiable => can communicate with both supervisor API endpoints on device or Resin API endpoints.
+    # _on_device = False, if RESIN_SUPERVISOR_ADDRESS and RESIN_SUPERVISOR_API_KEY env vars are not avaiable => can only communicate with Resin API endpoints.
+    if RESIN_SUPERVISOR_ADDRESS is None or RESIN_SUPERVISOR_API_KEY is None:
+        _on_device = False
+    else:
+        _on_device = True
+
+    def __init__(self):
+        self.base_request = BaseRequest()
+        self.settings = Settings()
+
+    def _check_args(self, device_id, app_id):
+        if device_id is None:
+            exceptions.MissingOption('device_id')
+
+        if app_id is None:
+            exceptions.MissingOption('app_id')
+
+    def _do_command(self, endpoint, req_data={}, device_id=None, app_id=None, method=None):
+        if not self._on_device:
+            self._check_args(device_id, app_id)
+
+            if req_data:
+                data = {
+                    'deviceId': device_id,
+                    'appId': app_id,
+                    'data': req_data
+                }
+            else:
+                data = {
+                    'deviceId': device_id,
+                    'appId': app_id
+                }
+
+            return self.base_request.request(
+                'supervisor/{0}'.format(endpoint),
+                'POST',
+                data=data,
+                endpoint=self.settings.get('api_endpoint'),
+                login=True
+            )
+        else:
+            return self.base_request.request(
+                endpoint,
+                method,
+                data=req_data,
+                endpoint=self.RESIN_SUPERVISOR_ADDRESS,
+                api_key=self.RESIN_SUPERVISOR_API_KEY
+            )
+
+    def force_api_endpoint(self, endpoint):
+        """
+        Force all API commands to a specific endpoint.
+
+        Args:
+            endpoint (bool): True if selecting the API on device. False if selecting the Resin API proxy endpoint.
+
+        Raises:
+            InvalidOption: if endpoint value is not bool.
+
+        """
+
+        try:
+            self._on_device = bool(endpoint)
+        except ValueError:
+            exceptions.InvalidOption('endpoint')
```

Comment:
@dfunckt  this is the only Supervisor instance so no need to worry about `_on_device`. Removed catching exeception